### PR TITLE
Add properties for content security policy

### PR DESCRIPTION
**Pull request by qerub**
I had some time over so I went ahead and made a patch for #7357 at risk. Hope you like it! 😀

### CODE REVIEW

**Comment by philwebb** [comment]
> Hope you like it! 😀
I do :)

**Comment by philwebb** [comment]
It might be worth changing `security.headers.content-security-policy-report-only` to an enum. `security.headers.content-security-policy-mode=default/report-only`.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SpringBootWebSecurityConfiguration.java [hunk 1]
```diff
@@ -108,6 +108,15 @@ public static void configureHeaders(HeadersConfigurer<?> configurer,
 		if (!headers.isContentType()) {
 			configurer.contentTypeOptions().disable();
 		}
+		if (StringUtils.hasText(headers.getContentSecurityPolicy())) {
+			if (headers.getContentSecurityPolicyMode() == Headers.ContentSecurityPolicyMode.DEFAULT) {
+				configurer.contentSecurityPolicy(headers.getContentSecurityPolicy());
+			}
+			else {
+				assert headers.getContentSecurityPolicyMode() == Headers.ContentSecurityPolicyMode.REPORT_ONLY;
```
Comment (qerub) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SpringBootWebSecurityConfiguration.java]: Is this kind of `assert` in line with your coding standards?

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java [hunk 2]
```diff
@@ -174,6 +174,18 @@ public void setFilterDispatcherTypes(Set<String> filterDispatcherTypes) {
...
+			/**
+			 * Use the {@code Content-Security-Policy} header.
+			 */
+			DEFAULT,
+
+			/**
+			 * Use the {@code Content-Security-Policy-Report-Only} header.
+			 */
+			REPORT_ONLY
+		}
```
Comment (qerub) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java]: Enum added!

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SpringBootWebSecurityConfiguration.java (hunk 1 above)
Comment (philwebb) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SpringBootWebSecurityConfiguration.java]: Afraid not, we generally use Spring's `Assert` class. I can change this when we merge it.

**Pull request closed without merging**